src/Repository/FetcherConfigurationRepository.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Repository;
+
+use App\Entity\FetcherConfiguration;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Persistence\ManagerRegistry;
+
+/**
+ * @method FetcherConfiguration|null find($id, $lockMode = null, $lockVersion = null)
+ * @method FetcherConfiguration|null findOneBy(array $criteria, array $orderBy = null)
+ * @method FetcherConfiguration[]    findAll()
+ * @method FetcherConfiguration[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class FetcherConfigurationRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, FetcherConfiguration::class);
+    }
+
+    public function findOneOrCreate(string $name): FetcherConfiguration
+    {
+        $config = $this->findOneBy(['fetcherName' => $name]);
+
+        if (null === $config) {
+            $config = new FetcherConfiguration();
+            $config->setFetcherName($name);
+            $this->_em->persist($config);
+            $this->_em->flush();
```

Comment:
Do not persist + flush here.

Usually we should never flush outside an entry point (controller, command, message handler). => actually follow **always** this advice

Flush automatically save everything that is in the UOW (Unit Of Work). So here one may think that only the fetcher will be saved, but depending on the previous executed code, many other things could be save. This is a source of bug and hard time to debug.